subsys/logging/log_msg.c

```diff
@@ -95,10 +95,14 @@ u32_t log_msg_arg_get(struct log_msg *msg, u32_t arg_idx)
 
 const char *log_msg_str_get(struct log_msg *msg)
 {
-	if (log_msg_nargs_get(msg) <= LOG_MSG_NARGS_SINGLE_CHUNK) {
-		return msg->data.data.std.str;
+	if ((log_msg_is_std(msg) &&
+		(log_msg_nargs_get(msg) > LOG_MSG_NARGS_SINGLE_CHUNK)) ||
+	    (!log_msg_is_std(msg) &&
+		(msg->hdr.params.hexdump.length >
+			LOG_MSG_HEXDUMP_BYTES_SINGLE_CHUNK))) {
+		return msg->payload.ext.str;
```

Comment:
Could we put str as common field for all message types?

Reply:
done